modules/raft/src/main/java/org/apache/ignite/raft/jraft/util/internal/ThrowUtil.java

```diff
@@ -46,17 +38,50 @@ public static void throwException(final Throwable t) {
         throw (E) t;
     }
 
-    public static <T extends Throwable> T cutCause(final T cause) {
-        Throwable rootCause = cause;
-        while (rootCause.getCause() != null) {
-            rootCause = rootCause.getCause();
-        }
+    /**
+     * Checks if passed in {@code 'Throwable'} has given class in {@code 'cause'} hierarchy
+     * <b>including</b> that throwable itself.
+     * <p>
+     * Note that this method follows includes {@link Throwable#getSuppressed()}
+     * into check.
+     *
+     * @param t Throwable to check (if {@code null}, {@code false} is returned).
+     * @param msg Message text that should be in cause.
+     * @param cls Cause classes to check (if {@code null} or empty, {@code false} is returned).
+     * @return {@code True} if one of the causing exception is an instance of passed in classes,
+     *      {@code false} otherwise.
+     */
+    @SafeVarargs
+    public static boolean hasCause(@Nullable Throwable t, @Nullable String msg, @Nullable Class<?>... cls) {
```

Comment:
`@Nullable Class<?>... cls` - this notation actually means "an array of Class instances that can *contain* nulls", which doesn't make a lot of sense in this method

Reply:
The annotation is used to indicate the `null `is a valid value for the cls array parameter, so it makes sense to me.

Reply:
If you want to indicate that `null` is a valid value for the whole array, you should write `Class<?> @Nullable... cls` instead